galasa-inttests-parent/dev.galasa.inttests/src/main/java/dev/galasa/inttests/compilation/AbstractCompilationLocal.java

```diff
@@ -0,0 +1,411 @@
+                + " -d " + remoteGradleDir.toString()
+                + "; echo $?"
+        );
+        
+        logger.trace("Checking return code of unzip");
+        assertThat(unzipRC).isEqualToIgnoringWhitespace("0");
+        
+        Path gradleWorkingDir = remoteGradleDir.resolve("gradle-" + gradleZipVersion + "/bin");
+        
+        logger.trace("Checking unpacked gradle version");
+        String gradleVersion = getLinuxImage().getCommandShell().issueCommand(
+                javaHomeCommand + "; " +
+                gradleWorkingDir + "/gradle -v"
+        );
+        
+        assertThat(gradleVersion).contains(gradleZipVersion);
+        
+        return gradleWorkingDir;
+    }
+    
+    /*
+     * Retrieves the necessary source files from the unpacked simplatform archive and 
+     * structures them in a parent directory, along with a parent settings file.
+     * 
+     * @param   unpackedDir     The directory containing he unpackaged simplatform archive  
+     * @return  parentDir       The parent directory of the newly structured simplatform parent project
+     * 
+     */
+    private Path structureSimplatform(Path unpackedDir) throws IOException {
+        // Create new (temp) directory
+        logger.trace("Creating project parent directory (for manager and tests)");
+        Path parentDir = Files.createTempDirectory("galasa.test.simplatform.parent");
+        parentDir.toFile().deleteOnExit();
+        
+        // Create parent settings file
+        
+        logger.trace("Creating settings.gradle");
+        Path parentSettingsFile = parentDir.resolve("settings.gradle");
+        Files.createFile(parentSettingsFile);
+        StringBuilder settingsSB = new StringBuilder();
+        settingsSB.append("include '");
+        settingsSB.append(managerProjectName);
+        settingsSB.append("'\n");
+        settingsSB.append("include '");
+        settingsSB.append(testProjectName);
+        settingsSB.append("'\n");
+        Files.write(parentSettingsFile, settingsSB.toString().getBytes());
+        
+        // Get Manager Files
+        logger.trace("Copying managers source into parent directory");
+        copyDirectory(unpackedDir.resolve("simplatform-main/galasa-simbank-tests/" + managerProjectName + "/"), parentDir.resolve(managerProjectName));
+        // Get Tests
+        logger.trace("Copying tests source into parent directory");
+        copyDirectory(unpackedDir.resolve("simplatform-main/galasa-simbank-tests/" + testProjectName + "/"), parentDir.resolve(testProjectName));
+
+        return parentDir;
+    }
+    
+    /*
+     * Copy a directory to another directory using nio Files.
+     * 
+     * @param   source      The directory from which to copy files / directories  
+     * @param   target      The directory to which files / directories will be copied
+     * 
+     */
+    private void copyDirectory(Path source, Path target) throws IOException {
```

Comment:
not needed if you xfer the zip instead of individual files

Reply:
Files are now unpackaged on the remote image. This method has been removed.